src/executor.cc

```diff
@@ -111,6 +111,17 @@ Result Executor::Execute(Engine* engine,
       r = engine->DoClearDepth(cmd->AsClearDepth());
     } else if (cmd->IsClearStencil()) {
       r = engine->DoClearStencil(cmd->AsClearStencil());
+    } else if (cmd->IsCopy()) {
+      auto copy = cmd->AsCopy();
+      auto buffer_from = copy->GetBufferFrom();
+      auto buffer_to = copy->GetBufferTo();
+      if (buffer_from->IsDataBuffer() && buffer_to->IsDataBuffer()) {
```

Comment:
I don't think you need to do this. C++ should pick the overridden method by default so just doing `buffer_from->CopyTo(buffer_to)` should be fine I believe.

Reply:
Thanks, and apologies for all these newbie bumps, I managed to stick to C so far in my career so I am a bit walking on eggs when dealing with C++ inheritance and overriding.